ctk/features/try.feature

```diff
@@ -0,0 +1,81 @@
+Feature: Try Task
+  As an implementer of the workflow DSL
+  I want to ensure that try tasks can be executed within the workflow
+  So that my implementation conforms to the expected behavior
+
+  # Tests that try tasks complete when catching errors, and execute the defined handler task, if any
+  # Tests simple uri interpolation
+  # Tests custom error variable name
+  # Tests error instance path
+  Scenario: Try Handle Caught Error
+    Given a workflow with definition:
+    """yaml
+    document:
+      dsl: 1.0.0-alpha1
+      namespace: default
+      name: try-catch-404
+    do:
+      tryGetPet:
```

Comment:
I see that we are using names for some "dos", are those optional or mandatory? 
And which is the purpose? 
I see "tryGetPet" name convenient to track progress of the workflow. With that spirit, probably when waiting for an event, the name should be mandatory (to be able to report where the workflow is actually waiting)

Reply:
what the $err object will contain if a do in a for fails?

Reply:
You mean, as `instance`? Yeah, I forgot to document that.

So far, in my runtime tests, I'm using the following convention: `/for/{index}/do`.

We could work with anything else, too. It just looked pretty to me 😆

Reply:
no, its fine. I was actually thinking if we have an alternative to optionally skip the task names overall (not just for foreach), so, when the task name is present we use it for error.instance, but if its not there, we use just a generated, but predictable, name. For example if the second task in the workflow throw an error and the task is unname, instance will be `/do/2 `

Reply:
It's believe a very valid point! 

I'm wondering if the auto generated task name is DSL specific or rather interpreter/UI specific stuff? 

In my opinion, when present in a **collection** of tasks (as in `workflow.do` or `execute.sequentially` or `execute.concurrently`), the name should be required, in whichever way is most convenient to the interpreter/UI, as it's very opinated (some would maybe prefer naming it task-1 rather than task1 or even name it thanks to a zero based index, etc.).

Reply:
the problem I see there is you consider the following:

```yaml
do: 
  execute:
    name: composite # does not work, execute is a collection of tasks
    try: 
       name: tryWithGet
       call: 
    try1:  #we are force to use another name than try, which is already defined. but task1 is not a valid task type.
       name: tryWithGet1
       call: 
```

Reply:
this drives me to next point, if name is optional, execute should be an array, not an object ;)

Reply:
but we can discuss on a different issue

Reply:
Thats intuitive for me, if user does not write a name, instance is /do/execute/1 (that predictable generated name), if user wants a name there, he can write it

Reply:
Anyway, lets close this (in current spec user is forcer to write names for anything anyway), Ill open an issue to track it